csr/csr.go

```diff
@@ -66,6 +66,17 @@ func VerifyCSR(ctx context.Context, csr *x509.CertificateRequest, maxNames int,
 	if !features.Enabled(features.SHA1CSRs) && csr.SignatureAlgorithm == x509.SHA1WithRSA {
 		return unsupportedSigAlg
 	}
+
+	if features.Enabled(features.RejectDuplicateCSRExtensions) {
+		oidSeen := make(map[string]bool)
+		for _, ext := range csr.Extensions {
+			if oidSeen[ext.Id.String()] {
+				return berrors.MalformedError("extension OID %s appears in CSR twice. File an issue with the maintainer of your ACME client.", ext.Id)
+			}
+			oidSeen[ext.Id.String()] = true
+		}
+	}
```

Comment:
Optionally, we could finish looping and construct a message that contains a list of the OIDs listed two or more times.

Reply:
I'd prefer to keep this simple for now -- the less logic there is surrounding ASN.1 spelunking the better, imo.